### DISCUSSION TITLE
yaml input to create a blade in iVABS 0.11

### DISCUSSION THREAD

**Discussion by rdrclrbh** [wenbinyugroup/.github#36] [VABS Related Discussions] [answered]
I have the yaml input for creating a blade with "analysis" for V0.8 but I cannot find an example on how to do it with the new yaml format. And the naming looks to be different. V 0.8 uses "structure_class", but now it looks like "type" is used. Furthermore what used to be a "blade" structure_class does not appear to be carried over as a type, i.e., type: "blade" does not work. 

Is there an example that shows how to replace generating blade cross sections the way it was done in V0.8 with the new syntax in V0.11?

**Comment by unitasium** [comment]
Please check the attached files.
[new_input.zip](https://github.com/user-attachments/files/18221986/new_input.zip)

**Reply by vacupenda** [reply]
Hello, could you please provide the complete set of run files? I would like to study the new version of the optimization program, but I often encounter errors. If I have a complete set of run files, I believe it could help resolve my current issues.

**Reply by unitasium** [reply]
You can find examples using the new input format in the release page: https://github.com/wenbinyugroup/ivabs/releases/tag/v0.11

**Reply by rdrclrbh** [reply]
This does not quite resolve my issues setting this up. The first thing I am trying to figure out is how to specify use of more than one template along the length of the blade. In V0.8 it was done with

design:
  - name: "blade_1"
    structure_class: "blade"
    cross_section_base_design: |
      0, airfoil_rbox_uni.xml.tmp
      0.95500, airfoil_nospar_uni.xml.tmp
      
    How do I do that in V0.11? Once we get that sorted out, I can see if there are other issues that crop up. Thanks

**Reply by unitasium** [reply]
Please check out this answer: https://github.com/orgs/wenbinyugroup/discussions/40#discussioncomment-11716404

**Reply by rdrclrbh** [reply]
I seem to be making some progress, now that I have a blade template distribution. But the next problem I am encountering is there is no longer a "blade" analysis type is appears. i.e., how does one handle the functionality that used to be addressed with this yaml block:

analysis:
  steps:
    - step: "generate cs design"
      type: "blade"
      blade_1:
        settings:
          analysis: 1
        inputs:
          stations: [0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.9371, 1.0]

**Reply by unitasium** [reply]
This step is not needed and its functionality is now part of the homogenization step:

````python
analysis:
  steps:
    - step: "cs analysis"
      type: "cs"
      analysis: 'h'
````

**Reply by rdrclrbh** [reply]
Hmmm, then I don't see how one specifies where to evaluate each cross-section along the length of the blade. In V0.8 one could specify the station location as a "stations" variable to specify that ivabs create a cross-section at each specified station using the specified template and parameter values (from given distributions). I see no way from the provided example yaml or data_proc_funcs_v0.11.py to provide to ivabs the stations where I want cross-sections constructed and evaluated. Is this specified somewhere else now?

**Reply by unitasium** [reply]
It is specified in the `structure>model` section.

If you want to use the blade nodes defined in your rcas file, you can do this:
````ymal
structure:
  model:
    main_file: 'rcas_input.dat'
    tool: 'rcas'  # FENODE > Blade nodes
    config:
      node_id_start: 2
````

If you want to specify the cs locations explicitly:
````ymal
structure:
  model:
    section_locations:
      type: 'curvilinear'
      coordinates: [0, 6, 15, 27, 30]
````

For further details, you can check these two examples: `anl_blade_cs_rcas` and `anl_distr_cs_airfoil`.

**Reply by rdrclrbh** [reply]
Ah, I see that clears things up considerably. I'll give it a try.

so if I do using tool: rcas, and node_id_start: 2, will is start at node 2 and then create x-sections at all of the subsequent nodes listed in the rcas file?

**Reply by unitasium** [reply, marked as answer]
Correct.
